scripts/release/build_release.js

```diff
@@ -0,0 +1,88 @@
+import { execSync } from 'child_process'
+import prompt from 'prompt'
+import semver from 'semver'
+import { version as packageVersion } from '../../package.json'
+
+import {
+  checkError,
+  confirmResponsePattern,
+  isPositiveResponse,
+} from './utils'
+
+prompt.message = 'Confirm'
+
+const upVersion = ({ versioningType, packageVersion }) => {
+  const major = semver.major(packageVersion)
+  const minor = semver.minor(packageVersion)
+  const patch = semver.patch(packageVersion)
+
+  switch(versioningType) {
+    case 'patch': {
+      return `${major}.${minor}.${parseInt(patch) + 1}`
+    }
+    case 'minor': {
+      return `${major}.${parseInt(minor) + 1}.${patch}`
+    }
+    case 'major': {
+      return `${parseInt(major) + 1}.${minor}.${patch}`
+    }
+    default: {
+      throw `bad npm versioning type! Acceptable types: patch, minor, major. Type passes: ${versioningType}`
+    }
+  }
+}
+
+const confirmBuild = {
+  name: 'This will create a git branch, build the package, up the package version and commit/push the results to the Snacks repo. Are you sure you want to build a new release of Snacks?',
+  type: 'string',
+  pattern: confirmResponsePattern,
+  required : true,
+  default: 'yes'
+}
+
+const confirmBuildCheck = (userResponse) => {
+  if (!isPositiveResponse(userResponse)) {
+    console.log('Build confirmation failed. Exiting build...')
+    return false
+  }
+
+  return true
+}
+
+const confirmVersion = {
+  name: 'Is this a patch, minor or major version?',
+  type: 'string',
+  pattern: /patch|Patch|minor|Minor|Major|major|/,
```

Comment:
looks like an extra `|` was added at the end

Reply:
also may be worth adding `^` and `$`

something like `/^\s*(?:[pP]atch|[mM]inor|[mM]ajor)\s*$/`

Reply:
or just `/^\s*(?:major|minor|patch)\s*$/i`